theano/tensor/nnet/corr_gemm.c

```diff
@@ -172,50 +171,61 @@ PyArrayObject* corrMM(PyArrayObject* bottom,
     }
 
     // Create temporary columns
-    npy_intp col_dim[2];
-    col_dim[0] = (npy_intp)(nChannels * kW * kH);
-    col_dim[1] = (npy_intp)(topHeight * topWidth);
-    PyArrayObject* col = (PyArrayObject*)PyArray_EMPTY(2,
-		                           col_dim,
-                                           PyArray_TYPE(top),
-					   0);
+    const int max_threads = %(cores)s < batchSize ? %(cores)s : batchSize;
```

Comment:
why not use omp_get_max_threads(); instead of core? I have the impression that using core will bypass the env variable OMP_NUM_THREADS=N. If we can keep that working, it would be good. It would also remove the dependency on multiprocessing.